src/model.jl

```diff
@@ -1,3 +1,85 @@
+"""
+    TapedModel{F}
+
+State wrapper to hold `Libtask.CTask` model initiated from `f`
+"""
+struct TapedModel{F}
+    f::F
+    ctask::Libtask.TapedTask
```

Comment:
IIRC `TapedTask` is not a concrete type anymore. Should we use a type parameter here?